services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/manage_units/__tests__/unit.test.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Unit from '../unit';
+
+const mockData = {
+  classroomActivities: new Map([
+    [
+      '1',
+      {
+        name: 'Activity 1',
+        completed: false,
+        started: false,
+        studentCount: 10,
+        ownedByCurrentUser: true,
+        ownerName: 'Teacher A',
+        activityId: 'a1',
+        classroom_unit_id: 'cu1',
+      },
+    ],
+    [
+      '2',
+      {
+        name: 'Activity 2',
+        completed: true,
+        started: true,
+        studentCount: 0,
+        ownedByCurrentUser: false,
+        ownerName: 'Teacher B',
+        activityId: 'a2',
+        classroom_unit_id: 'cu2',
+      },
+    ],
+  ]),
+  classrooms: ['Class 1', 'Class 2'],
+  num_students_assigned: 20,
+  unitName: 'Sample Unit',
+  unitId: 'unit1',
+};
+
+describe('Unit Component', () => {
+  it('renders the unit name and assigned information', () => {
+    render(<Unit data={mockData} />);
+
+    expect(screen.getByText('Sample Unit')).toBeInTheDocument();
```

Comment:
This is something @anathomical often calls out, but I think that where something is a reference to a variable (as opposed to static text on the page), we should try to refer to it that way instead of just as a string, so it's clear what it should be testing.

Reply:
so something like this? 

```
const unitNameSpan = screen.getByText('Sample Unit')
expect(screen.getByText(unitNameSpan)).toBeInTheDocument();
```